### ISSUE TITLE
Conv1D formula in docs is wrong

### ISSUE THREAD

**Issue by Vichoko** [pytorch/pytorch#47357]
## 📚 Documentation

in [torch.nn.Conv1d](https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html) docs the formula that describe the 1D convolution is wrong.

The first symptom that something is odd is that if you compare the formula with [torch.nn.Conv2d](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html) it's exactly the same.

Now I'm gonna explain why the Conv1d formula is wrong. Please take a look at the formula exposed in [torch.nn.Conv1d docs](https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html).

For simplicity, take in count that N=1 and bias=0.
So for an arbitrary out-channel C_out_j, the value it takes it's the product of the linear combination of all the C_in input channels convolutions on the kernel. 

This combination is wrong because in 1D convolution all the C_in channels should be independent (this is precisely the difference between 1d and 2d convolution). So because of this, the output channel C_out_j should depend only on a single C_in channel and there is no way the output value could depend on 2 or more C_in channels.

**Comment by Vichoko** [comment]
Anyone can explain why different channels convolutions are combined through a sum?
I cant understand the intuition behind the combination of input channels outputs.

For example, if you are modeling 3 different and pseudo-independent signals, for example, temperature, ambient sound volume, and vibration of the earth. And you 1D Convolute over them, the output of the convolutions of each signal will be summed on the same output channel. I just feel this is completely wrong. The intuition says that these independant signals should be treated independently, so the convolution over them shouldn't be linearly combined at any time. 

But in runtime, it would be efficient for these independent 1d signals to be treated at the same time.

**Comment by Vichoko** [comment]
Well, I just noticed the `groups` parameters control this channel combination, so in the code, I can make it work.
I still think the formula shown in the docs should reflect this variation because 1d Conv isn't just a 2d Conv with H or W set to 1.

**Comment by SsnL** [comment]
I don't think that the formula is wrong. It is the same as 2d formula because the difference in dimensionality is reflected in star operator, yet the general form is valid for all dimensionalities.

**Comment by Vichoko** [comment]
Yeah, the formula isn't wrong. The observation I did is that the formula only works when all inputs are convolved to all outputs, which is the default behaviour explained in the docs and it's used in many 1D applications.

I noted that the formula doesn't work when using a different convolution scheme, like the one I explained when you don't want to convolve and linearly combine all channels. This behaviour is changed through the `groups`  parameter. At least in my case, I tried to explain the scheme where `groups=in_channels`. In this case, the formula changes to:

output(N_{i},  C_{out_{j}}) = weight(C_{kernel_{l}}) \star input(C_{in_{m}})

Where C_{kernel} = \floor{C_{out} / C_{in}}, for all j between 1 and C_{out}. Where l = (j - 1) mod C_{kernel} + 1 and m = \ceil{j / C_{kernel}}

**Comment by mruberry** [comment]
Our convolution documentation is pretty poor and we will accept PRs improving it. It would be nice, however, if any improvement to convolution documentation was holistic. That is, it considered all the convolution operations and made their documentation clear, precise, and consistent.
